sympy/matrices/common.py

```diff
@@ -2218,37 +2223,50 @@ def __neg__(self):
         return self._eval_scalar_mul(-1)
 
     @call_highest_priority('__rpow__')
-    def __pow__(self, num):
+    def __pow__(self, exp):
         if self.rows != self.cols:
             raise NonSquareMatrixError()
         a = self
-        jordan_pow = getattr(a, '_matrix_pow_by_jordan_blocks', None)
-        num = sympify(num)
-        if num.is_Number and num % 1 == 0:
+        jordan_pow = getattr (a, '_matrix_pow_by_jordan_blocks', None)
+        exp = sympify(exp)
+
+        if exp.is_zero:
+            return a._new(a.rows, a.cols, lambda i, j: int(i == j))
```

Comment:
I think that `return a.eye(a.rows)` is better here.

Reply:
AttributeError: 'ArithmeticOnlyMatrix' object has no attribute 'eye'

Reply:
AttributeError: 'EigenOnlyMatrix' object has no attribute '_mat' from somewhere if I try to use the global `eye()`

Reply:
This line is fine then (maybe the eye method should be added to the other classes though...)